src/script/interpreter.cpp

```diff
@@ -1286,6 +1285,95 @@ bool EvalScript(std::vector<valtype> &stack, const CScript &script,
                         std::reverse(data.begin(), data.end());
                     } break;
 
+                    case OP_RAWLEFTBITSHIFT: {
+                        // (in bits -- out)
+                        if (stack.size() < 2) {
+                            return set_error(
+                                serror, ScriptError::INVALID_STACK_OPERATION);
+                        }
+                        valtype &data = stacktop(-2);
+                        const int32_t signed_bitshift =
+                            CScriptNum(stacktop(-1), fRequireMinimal).getint();
+                        popstack(stack);
+                        if (data.empty() || signed_bitshift == 0) {
+                            break;
+                        }
+                        uint32_t bitshift = abs(signed_bitshift);
+                        if (bitshift > data.size() * 8) {
+                            bitshift = data.size() * 8;
+                        }
+                        const size_t inner_bitshift = bitshift % 8;
+                        const size_t byteshift = bitshift / 8;
+                        if (signed_bitshift > 0) { // Shift left
+                            const uint8_t mask = 0xff >> inner_bitshift;
+                            // Copy bits over to the destination
+                            for (size_t idx = byteshift; idx < data.size();
```

Comment:
Could you drop the hanging conditionals at the end, and instead let this go to `data.size() + byteshift` and then define bits as: `const uint8_t bits = idx < data.size() ? data[idx] : 0;`